pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarMetadata.java

```diff
@@ -273,14 +278,24 @@ public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSess
         return columns.build();
     }
 
-    private ConnectorTableMetadata getTableMetadata(SchemaTableName schemaTableName, boolean withInternalColumns) {
+    @Override
+    public void cleanupQuery(ConnectorSession session) {
+        if (pulsarConnectorConfig.getAuthorizationEnable()) {
+            pulsarAuth.cleanSession(session);
+        }
+    }
+
+    private ConnectorTableMetadata getTableMetadata(ConnectorSession session, SchemaTableName schemaTableName,
+                                                    boolean withInternalColumns) {
 
         if (schemaTableName.getSchemaName().equals(INFORMATION_SCHEMA)) {
             return null;
         }
 
         TopicName topicName = getMatchedTopicName(schemaTableName);
 
+        checkTopicAuthorization(session, topicName.toString());
```

Comment:
Maybe we add auth check here is enough, other methods are all based on this method. The table name in the query will convert to a topic name, and we need to use the topic name to check the authorization.

Reply:
You're right. Thanks!

Reply:
There is an interface method `getTableHandle` that doesn't go through the `getTableMetadata` method. So I added an auth check to the `getTableHandle`